fetch/mesh_data.go

```diff
@@ -232,9 +233,7 @@ func (f *Fetch) GetMaliciousIDs(ctx context.Context, peer p2p.Peer) ([]types.Nod
 		if err := f.meteredStreamRequest(
 			ctx, malProtocol, peer, []byte{},
 			func(ctx context.Context, s io.ReadWriter) (int, error) {
-				return server.ReadResponse(s, func(respLen uint32) (n int, err error) {
-					return codec.DecodeFrom(s, &malIDs)
-				})
+				return readIDSlice(s, &malIDs.NodeIDs, maxMaliciousIDs)
```

Comment:
Would this also work?

```go
return server.ReadResponse(rw, func(respLen uint32) (n int, err error) {
	if respLen > maxMeshHashes*types.Hash32Length {
		return 0, errors.New("meshHashesStream: response too big")
	}
	return codec.DecodeFrom(io.LimitReader(rw, int64(respLen)), &malIDs)
})
```

Reply:
As discussed: `readIDSlice` also verifies consistency between outer `Response` length and the slice length